mlflow/pyspark/optuna/study.py

```diff
@@ -239,13 +241,15 @@ def run_task_on_executor_pd(iterator):
             self.mlflow_client.set_terminated(self._study_id, "KILLED")
             raise
         if "error" in result_df.columns:
-            error_count = result_df.filter(col("error") != "").count()
+            failed_runs = result_df.filter(col("error").isNotNull())
+            error_count = failed_runs.count()
             if error_count > 0:
-                first_non_null_value = result_df.select(first("error", ignorenulls=True)).first()[0]
```

Comment:
One issue:

`failed_runs.count()` and `failed_runs.select(...).first()` will spawn 2 Spark jobs (the second job is doing recomputation, which wastes resources), we can avoid it by:

```
error_rows = failed_runs.select("error").collect()
if len(error_rows) > 0:
   first_non_null_value = error_rows[0][0]
```